lib/pdf/core/page.rb

```diff
@@ -167,6 +173,9 @@ def init_new_page(options)
         @dictionary = document.ref(:Type        => :Page,
                                    :Parent      => document.state.store.pages,
                                    :MediaBox    => dimensions,
+                                   :CropBox     => dimensions,
+                                   :BleedBox    => dimensions,
```

Comment:
Yeah, the printer's software wouldn't let me leave them out in order to validate.  In an ideal world they would follow the fallbacks as you described.   I researched this topic for a couple days and found no ill effects of including them for compatibility.